### PR TITLE
cleanup: Update qtox windows builder image to Qt 6.8.1.

### PR DESCRIPTION
**Pull request by iphydf**
We don't need to download the entire Qt source code. We only use 3 modules: qtbase, qttools, qtsvg. Also, parallelised the 3 main sections of the image build: Qt, non-Qt deps, and gdb.

Also, upgraded the Windows Qt version to 6.8.1. This is possible because of our host-qt image, which has the latest Qt built for Linux.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/dockerfiles/212)

### CODE REVIEW

File: qtox/build_qt_windows.sh [hunk 1]
```diff
@@ -15,42 +15,80 @@ parse_arch --dep "qt" --supported "win32 win64" "$@"
...
-
 CROSS_COMPILE="$MINGW_ARCH-w64-mingw32-"
 "${CROSS_COMPILE}gcc" --version
 
-./configure -prefix "$DEP_PREFIX" \
+mkdir qtbase/_build && pushd qtbase/_build
+../configure -prefix "$DEP_PREFIX" \
+  -appstore-compliant \
   -release \
-  -shared \
```
Comment (nurupo) [inline qtox/build_qt_windows.sh:26]: What is the behavior if we don't specify "-shared"?

File: qtox/build_qt_windows.sh [hunk 2]
```diff
@@ -15,42 +15,80 @@ parse_arch --dep "qt" --supported "win32 win64" "$@"
...
+  -no-feature-sql \
+  -no-feature-dbus \
+  -no-opengl \
   -device-option "CROSS_COMPILE=$CROSS_COMPILE" \
-  -qt-host-path /usr/lib/x86_64-linux-gnu/cmake \
+  -qt-host-path /work/host/qt \
   -platform linux-g++ \
   -xplatform win32-g++ \
-  -openssl "$(pkg-config --cflags --libs openssl)" \
-  -openssl-linked \
```
Comment (nurupo) [inline qtox/build_qt_windows.sh:32]: How come `-openssl-linked` was removed?

File: qtox/build_qt_windows.sh (hunk 1 above)
Comment (iphydf) [reply qtox/build_qt_windows.sh:26]: Added it back. It's the same, but I removed it while testing -static and forgot to put it back.

File: qtox/build_qt_windows.sh (hunk 2 above)
Comment (iphydf) [reply qtox/build_qt_windows.sh:32]: With Qt 6.8.1, I was having issues getting the linked version to find the library at runtime. With runtime-resolved libssl, it works. I'll have another look to see if I can get it working.

Comment (iphydf) [reply qtox/build_qt_windows.sh:32]: I've put it back, it works. I think what fixed it was qt-cmake.

**Review by Green-Sky — APPROVED** [review]

**Review by nurupo — APPROVED** [review]

**Pull request merged**